src/projects/monitoring/alert/notification.cpp

```diff
@@ -0,0 +1,169 @@
+		{
+			auto notification = std::make_shared<Notification>();
+
+			notification->_notification_server_url = notification_server_url;
+			notification->_timeout_msec = timeout_msec;
+			notification->_secret_key = secret_key;
+			notification->_source_uri = source_uri;
+			notification->_message_list = message_list;
+			notification->_stream_metric = stream_metric;
+			notification->_status_code = StatusCode::OK;
+
+			notification->Run();
+			
+			return notification;
+		}
+
+		Notification::StatusCode Notification::GetStatusCode() const
+		{
+			return _status_code;
+		}
+
+		ov::String Notification::GetErrReason() const
+		{
+			return _err_reason;
+		}
+
+		uint64_t Notification::GetElapsedTime() const
+		{
+			return _elapsed_ms;
+		}
+
+		void Notification::SetStatus(StatusCode code, ov::String reason)
+		{
+			_status_code = code;
+			_err_reason = reason;
+		}
+
+		ov::String Notification::GetMessageBody()
+		{
+			// Make request message
+			Json::Value jv_root;
+
+			// Messages
+			Json::Value jv_messages;
+			if (_message_list == nullptr || _message_list->size() <= 0)
+			{
+				// If the message_list is empty, it means that the status of the stream has become normal, so send an OK message.
+				
+				Json::Value jv_message;
+
+				jv_message["code"] = Message::StringFromMessageCode(Message::Code::OK).CStr();
+				jv_message["description"] = Message::DescriptionFromMessageCode<bool>(Message::Code::OK, true, true).CStr();
+
+				jv_messages.append(jv_message);
+			}
+			else
+			{
+				for (auto &message : *_message_list)
+				{
+					Json::Value jv_message;
+
+					jv_message["code"] = Message::StringFromMessageCode(message->GetCode()).CStr();
+					jv_message["description"] = message->GetDescription().CStr();
+
+					jv_messages.append(jv_message);
+				}
+			}
+
+			// Metric infos
+			Json::Value jv_source_info = ::serdes::JsonFromStream(_stream_metric);
+
+			jv_root["sourceUri"] = _source_uri.CStr();
+			jv_root["messages"] = jv_messages;
+			jv_root["sourceInfo"] = jv_source_info;
+
+			return ov::Converter::ToString(jv_root);
+		}
+
+		void Notification::ParseResponse(const std::shared_ptr<ov::Data> &data)
+		{
+			ov::JsonObject object = ov::Json::Parse(data->ToString());
+			if(object.IsNull())
+			{
+				SetStatus(StatusCode::INVALID_DATA_FORMAT, ov::String::FormatString("Json parsing error : a response in the wrong format was received."));
+				return;
+			}
+		}
+
+		void Notification::Run()
+		{
+			auto body = GetMessageBody();
+			if(body.IsEmpty())
+			{
+				// Error
+				return;
+			}
+
+			// Set X-OME-Signature
+			auto md_sha1 = ov::MessageDigest::ComputeHmac(ov::CryptoAlgorithm::Sha1, _secret_key.ToData(false), body.ToData(false));
+			if(md_sha1 == nullptr)
+			{
+				// Error
+				SetStatus(StatusCode::INTERNAL_ERROR, ov::String::FormatString("Signature creation failed.(Method : HMAC(SHA1), Key : %s, Body length : %d", _secret_key.CStr(), body.GetLength()));
```

Comment:
```GetErrReason()``` is a "public" API, and there is a concern that unintended usage might expose the key value to the external system. Please adjust the access level or modify it to ensure that the key value is not exposed